contracts/example/tests/test.rs

```diff
@@ -158,15 +242,45 @@ fn gmp_example() {
     // Executing message on destination app
     destination_app.execute(&source_chain, &message_id, &source_address, &payload);
 
-    assert_last_emitted_event(
-        &env,
-        &destination_app.address,
-        (
-            Symbol::new(&env, "executed"),
-            source_chain,
-            message_id,
-            source_address,
-        ),
-        (payload,),
+    let gas_token = setup_gas_token(&env, &user);
+    let amount = 1000;
+    let token_id =
+        setup_interchain_token_service_token(&env, &interchain_token_service_client, &user, 1000);
+    let destination_address = Bytes::from_hex(&env, "4F4495243837681061C4743b74B3eEdf548D56A5");
+    let data = Some(Bytes::from_hex(&env, "abcd"));
+
+    // Register destination chain as trusted
+    interchain_token_service_client
+        .mock_all_auths()
+        .set_trusted_chain(&destination_chain);
+
+    // Add authorization for the token transfer
+    source_app.mock_all_auths().send_token(
+        &user,
+        &token_id,
+        &destination_chain,
+        &destination_address,
+        &amount,
+        &data,
+        &gas_token,
+    );
+
+    // Test token receiving functionality
+    let received_token_id = BytesN::<32>::random(&env);
+    let received_token_address = Address::generate(&env);
+    let received_amount: i128 = 500;
+    let received_payload: Bytes = BytesN::<20>::random(&env).into();
+
+    let source_address_bytes = Bytes::from_slice(&env, source_address.to_string().as_bytes());
+
+    // Execute token receive message
+    destination_app.execute_with_interchain_token(
```

Comment:
Call ITS.execute instead. This will internally call the app.
see this test
https://github.com/axelarnetwork/axelar-cgp-soroban/blob/8239e4126cdccb4156f737dd6e20fad5c2bfc239/contracts/interchain-token-service/tests/executable.rs#L122